astropy/wcs/wcsapi/tests/test_high_level_wcs_wrapper.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from numpy.testing import assert_allclose
+
+from ....coordinates import SkyCoord
+
+from ..low_level_api import BaseLowLevelWCS
+from ..high_level_wcs_wrapper import HighLevelWCSWrapper
+
+
+class CustomLowLevelWCS(BaseLowLevelWCS):
+
+    @property
+    def pixel_n_dim(self):
+        return 2
+
+    @property
+    def world_n_dim(self):
+        return 2
+
+    @property
+    def world_axis_physical_types(self):
+        return ['pos.eq.ra', 'pos.eq.dec']
+
+    @property
+    def world_axis_units(self):
+        return ['deg', 'deg']
+
+    def pixel_to_world_values(self, *pixel_arrays):
+        return [np.asarray(pix) * 2 for pix in pixel_arrays]
+
+    def array_index_to_world_values(self, *index_arrays):
+        return [np.asarray(pix) * 2 for pix in index_arrays]
+
+    def world_to_pixel_values(self, *world_arrays):
+        return [np.asarray(world) / 2 for world in world_arrays]
+
+    def world_to_array_index_values(self, *world_arrays):
+        return [np.asarray(world) / 2 for world in world_arrays]
+
+    @property
+    def world_axis_object_components(self):
+        return [('test', 0, 'spherical.lon.degree'),
+                ('test', 1, 'spherical.lat.degree')]
+
+    @property
+    def world_axis_object_classes(self):
+        return {'test': (SkyCoord, (), {'unit': 'deg'})}
+
```

Comment:
Missing here is a of an *incomplete* `BaseLowLevelWCS` subclass, and testing that it fails (as it should) - I think that's worth adding at least a quick check of that given that we expect that to happen quite often if anyone actually subclasses this...

Reply:
Added